test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java

```diff
@@ -270,6 +279,58 @@ public ReduceContext forFinalReduction() {
         return namedXContents;
     }
 
+    @Override
+    protected NamedXContentRegistry xContentRegistry() {
+        return namedXContentRegistry;
+    }
+
+    @Override
+    protected final NamedWriteableRegistry getNamedWriteableRegistry() {
+        return namedWriteableRegistry;
+    }
+
+    /**
+     * Implementors can override this if they want to provide a custom list of namedWriteables.  If the implementor
+     * _just_ wants to register in namedWriteables provided by a plugin, prefer overriding
+     * {@link InternalAggregationTestCase#registerPlugin()} instead because that route handles the automatic
+     * conversion of AggSpecs into namedWriteables.
+     */
+    protected List<NamedWriteableRegistry.Entry> getNamedWriteables() {
+        List<NamedWriteableRegistry.Entry> entries
+            = new ArrayList<>(new SearchModule(Settings.EMPTY, emptyList()).getNamedWriteables());
```

Comment:
If you pass the plugins in instead of `emptyList()` then I think `SearchModule` will do this stuff for you.